test/reflection.jl

```diff
@@ -213,3 +213,10 @@ let t13464 = "hey there sailor"
         @test startswith(err13464.msg, "expression is not a function call, or is too complex")
     end
 end
+
+# Issue 14173
+module Tmp
+    export A
+    A = randn(2000, 2000)
+end
+@test @elapsed(whos(IOBuffer(), Tmp)) < 5
```

Comment:
actually this passes fine for me even without your change, so not sure this is really capturing the difference between `show` and `showlimited` very well